src/bin/main.rs

```diff
@@ -85,32 +87,33 @@ fn app(props: &AppProps) -> Html {
     // the ability to access and change its contents be mutable.
     let datapath = use_mut_ref(MipsDatapath::default);
 
+    let datapath_state = use_reducer(DatapathReducer::default);
```

Comment:
Do you know if there is a performance cost or benefit at all to using `use_reducer_eq` instead in this situation? Just curious since I know we're using `use_state_eq` everywhere.

Reply:
Once I get the communicator to avoid sending duplicate state updates, each update should trigger a rerender anyways so using `use_reducer` instead just avoids an equality check for something we'd already verified on the emulator core side. It would be better at the moment but the performance is fine for development and once we're done with v2 the advantage would go away.